src/generic_cube.rs

```diff
@@ -185,13 +185,56 @@ pub enum Move {
     Z(MoveVariant),
 }
 
+impl Move {
+    /// Extracts the MoveVariant of a Move.
+    pub fn get_variant(&self) -> MoveVariant {
```

Comment:
This is acceptable, but there might be a better way to do this. I'll come back to this with comments later.

Reply:
Other than restructuring the Move enum, there isn't.